src/query_parser/v2/parser.y

```diff
@@ -502,7 +502,7 @@ attribute(A) ::= ATTRIBUTE(B) COLON param_term(C). {
       value_len = found_value_len;
     }
   }
-  A = (QueryAttribute){ .name = B.s, .namelen = B.len, .value = value, .vallen = value_len };
+  A = (QueryAttribute){ .name = rm_strndup(B.s, B.len), .namelen = B.len, .value = value, .vallen = value_len };
```

Comment:
why dup?

Reply:
For vector attributes, we need the hold the name string in addition to the value string which is already held. But in second thought, we don't need the call `dup` here, but we can do it in `QueryVectorNode_ApplyAttribute` instead (and save `dup` for non-vector attributes). It will also allow us to revert the `rm_free((char*)$$.name);` calls that were added to free the inner name field in the `QueryAttribute` array...